apps/rush-lib/src/api/PackageJsonEditor.ts

```diff
@@ -72,6 +72,10 @@ export class PackageJsonEditor {
   // SemVer range in one of the other fields for consumers.  Thus "dependencies", "optionalDependencies",
   // and "peerDependencies" are mutually exclusive, but "devDependencies" is not.
   private readonly _devDependencies: Map<string, PackageJsonDependency>;
+
+  // NOTE: The "resolutions" field is a yarn specific feature that controls package
+  // resolution override within yarn.
+  private readonly _resolutions: { [name: string]: string };
```

Comment:
Should be a `Map<string, string>` to match what others are doing

Reply:
Or I guess, `Map<string, PackageJsonDependency>` if this matches a version specifier format?

Reply:
@D4N14L I agree that we should make this a complete PackageJsonEditor API rather than just a way to pass the options along.  A config-file-with-data is generally preferable to a script-with-code, so for everyday problems, `resolutions` is actually a better design than **pnpmfile.js**.  (Although **pnpmfile.js** is still a good feature for advanced problems.)

I thought about adapting the PNPM shim to apply `resolutions` for PNPM.  But it turns out that PNPM later implemented this same feature, but they call it `overrides`. (https://github.com/pnpm/pnpm/pull/2946)  Docs are [here](https://pnpm.io/package_json#pnpmoverrides).  In the issue https://github.com/pnpm/pnpm/issues/1221 they point out that NPM reinvented the same feature in their ["overrides" RFC](https://github.com/npm/rfcs/blob/isaacs/overrides/accepted/0000-overrides.md) but with some differences in its design.  So all 3 package managers implement this feature in slightly different ways, all using different **package.json** fields. 

We should support PNPM overrides and recommend it over **pnpmfile.js**.  But that's a separate topic from this PR.

Reply:
Fixed